packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -1707,6 +1708,105 @@ void main() {
     variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.android }),
   );
 
+  testWidgets('the selection behavior when clicking `Copy` item', (WidgetTester tester) async {
+    List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SelectableRegion(
+          focusNode: FocusNode(),
+          selectionControls: materialTextSelectionHandleControls,
+          contextMenuBuilder: (
+            BuildContext context,
+            SelectableRegionState selectableRegionState,
+          ) {
+            buttonItems = selectableRegionState.contextMenuButtonItems;
+            return const SizedBox.shrink();
+          },
+          child: const Text('How are you?'),
+        ),
+      ),
+    );
+
+    final RenderParagraph paragraph1 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('How are you?'), matching: find.byType(RichText)));
+    final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph1, 6)); // at the 'r'
+    addTearDown(gesture.removePointer);
+    await tester.pump(const Duration(milliseconds: 500));
+    // `are` is selected.
+    expect(paragraph1.selections[0], const TextSelection(baseOffset: 4, extentOffset: 7));
+
+    expect(buttonItems.length, 2);
+    expect(buttonItems[0].type, ContextMenuButtonType.copy);
+
+    // press `Copy` item
+    buttonItems[0].onPressed.call();
```

Comment:
Nit: Any advantage to doing it this way versus actually tapping the button?: `await tester.tap(find.text('Copy'));`

Reply:
thank you, actually I want to test the blew callback

lib/src/widgets/selectable_region.dart#L994
```
  List<ContextMenuButtonItem> get contextMenuButtonItems {
    return SelectableRegion.getSelectableButtonItems(
      selectionGeometry: _selectionDelegate.value,
      onCopy: () {
        _copy();

        // In Android copy should clear the selection.
        switch (defaultTargetPlatform) {
          case TargetPlatform.android:
            _clearSelection();
            break;
          case TargetPlatform.iOS:
            hideToolbar(false);
            break;
          case TargetPlatform.fuchsia:
          case TargetPlatform.linux:
          case TargetPlatform.macOS:
          case TargetPlatform.windows:
            break;
        }
      },
      onSelectAll: () {
        selectAll(SelectionChangedCause.toolbar);
      },
    );
  }
```

I tried to use `tester.tap(find.text('Copy'));`,it does call the blew function, but in this test case, I think I should test `contextMenuButtonItems` copy callback.

lib/src/widgets/selectable_region.dart#L1092
```
  @Deprecated(
    'Use `contextMenuBuilder` instead. '
    'This feature was deprecated after v3.3.0-0.5.pre.',
  )
  @override
  void copySelection(SelectionChangedCause cause) {
    _copy();
    _clearSelection();
  }
```

Reply:
Got it, I'm on board with that 👍